### DISCUSSION TITLE
Deploy to custom domain for dev deployment and use separate env variables

### DISCUSSION THREAD

**Discussion by F1r3w477** [vercel/community#5682] [Help] [answered]
### Summary

I'm trying to deploy my dev branch in github to dev.mywebsite.com (This part works great), however, I have environment variables set for the development environment that I want to be accessible to dev.mywebsite.com. The environment variables work correctly when set locally.

My current setup has ENV variables for prod/dev databases (mongodb), and is setup so prod deployments use the prod db, and vice versa. 

### Steps to Reproduce

Set customdomain.com linked to main
Set dev.customdomain.com linked to dev
Set separate env variables for prod and dev
Try and use dev env variables for dev.customdomain.com

**Comment by F1r3w477** [comment]
so I read that vercel treats all non prod environments as preview, so I updated my location to preview instead of development for the env variable, and it works fine, so it's something to do with getting my custom domain to be considered the development environment. Not sure how to do that, would appreciate insight.

**Comment by amyegan** [comment, marked as answer]
Hey @F1r3w477. You can assign a custom domain to a preview branch in the project's domain settings.

<img width="966" alt="image" src="https://github.com/vercel/community/assets/8247105/bdc00d3e-410a-476a-8f41-4b23dbcec746">


Don't forget that Development environment variables are only used in your development environment (typically local) and Preview env vars are used for Preview deployments. You can even set different Preview environment variables for specific branches.

https://vercel.com/docs/projects/environment-variables#preview-environment-variables

**Comment by F1r3w477** [comment]
Thank you! Didn't realize that development env variables were only used for local.
